src/main/java/org/folio/circulation/domain/UpdateItem.java

```diff
@@ -25,20 +31,27 @@ public UpdateItem(Clients clients) {
     itemsStorageClient = clients.itemsStorage();
   }
 
-  public CompletableFuture<Result<Item>> onCheckIn(
-    Item item,
-    RequestQueue requestQueue,
-    UUID checkInServicePointId) {
-
+  public CompletableFuture<Result<Item>> onCheckIn(Item item, RequestQueue requestQueue,
+      UUID checkInServicePointId, UUID loggedInUserId, DateTime dateTime) {
     return changeItemOnCheckIn(item, requestQueue, checkInServicePointId)
-      .after(updatedItem -> {
-        if(updatedItem.hasChanged()) {
-          return storeItem(updatedItem);
-        }
-        else {
-          return completedFuture(succeeded(item));
-        }
-      });
+      .next(addLastCheckInProperties(checkInServicePointId, loggedInUserId, dateTime))
+      .after(this::storeItem);
```

Comment:
Does this mean that the check that the item has changed is redundant?

If so, is that check performed anywhere else in the system?

Reply:
According to the requirements, the last checked in data should be updated on every check in regardless if the item has changed.
Speaking of other places in the system, I don't think so. There is one endpoint for check in, which is used by anyone, who wants to .... check  an item in... The storage is jsut storage.

Reply:
Apologies for the confusion, I meant, are there others places where `item.hasChanged()` is invoked? If not, that code can be removed.

Reply:
It's used in notices.